### DISCUSSION TITLE
Possible actions & tools to cope with limited storage space.

### DISCUSSION THREAD

**Discussion by CharlieSolan** [evercam/ex_nvr#376] [Ideas]
When a hard drive is nearly full (90%) can we start reducing the frame rate of the older recordings to free up space rather than replace the hard drive?

**Comment by gBillal** [comment]
Hi @CharlieSolan 
To drop the frame rate of already recorded videos you need first to decode the videos, drop frames let's say from 8fps to 4fps, re-encode the stream and delete the old recording. Unfortunately such a process is really expensive in terms of resources especially we're using 4K videos (we cannot even take advantage of the hardware encoder in the pi since it only support full HD).

Our alternatives is:

- Overwrite the old recordings (already implemented)
- Archive the old recordings by sending them to a remote storage like Aws S3 (we're working on this right now)

Other alternatives:
- Reduce the fame rate/bit rate/resolution on the camera config for the new footage
- Switch to H265+

The best alternative is to store the footage on a remote storage (the best quality footage is preserved and a space is freed)

**Reply by CharlieSolan** [reply]
My concern on the remote storage option is data connectivity - Our current snapshots to the cloud consume a large volume of data, would the option of remote storage mean additional data usage?

**Comment by marcoherbst** [comment]
I'll rename the title to be a little more general and allow us to gather other options. 

There are long term permanent solutions like 

- [ ] calculate HDD space initially,
- [x] stream settings @dhairyas17 link here to outline spec, please. Link: https://docs.evercam.io/doc/camera-configuration-zJVcwsg5c6


There may also be case by case actions, such as retrospectively deleting weekends or evenings to gain some time. Billal will make an API to delete recordings. 

- [ ]  provide an easy way for Techops to do custom deleting (as per billal's comments, deleting is probably of more interest than transcoding)
- [ ]  easy way to do cloud upload. (WIP)
